conda/cli/main_env_create.py

```diff
@@ -125,6 +125,7 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
         directory=os.getcwd(),
     )
     env = spec.environment
+    context.add_environment_file_config_source(args.file, env.get_configuration_settings())
```

Comment:
I like this solution to the configuration overrides! However we need to ensure it lands at the right time with CLI and env var sources. For example, in `conda env create -f env.yml -n mouse`, the env name is `mouse` not whatever `env.yml` carries. So CLI takes precedence. Same with `conda env create -f env.yml -n mouse -c defaults`; CLI takes precedence. Is this precedence order respected right now?

Reply:
linking to this other thread about this approach to adding config to the context https://github.com/conda/conda/pull/13928#discussion_r2005017049 

I haven't tried to see if precedence is preserved, but definitely will need to iron that out!